test/core/functional/pool_tests.js

```diff
@@ -1141,4 +1141,23 @@ describe('Pool tests', function() {
       pool.connect();
     }
   });
+
+  it('should support callback mode for connect', {
+    metadata: { requires: { topology: 'single' } },
+    test: function(done) {
+      const pool = new Pool(null, {
+        host: this.configuration.host,
+        port: this.configuration.port,
+        bson: new Bson()
+      });
+
+      pool.on('connect', () => done(new Error('connect was emitted')));
+      pool.connect(err => {
```

Comment:
Should we test that a connect event was not emitted?

Reply:
We do above, an error is passed to `done` if the event occurs.